packages/components/pie-modal/src/index.ts

```diff
@@ -255,13 +291,7 @@ export class PieModal extends RtlMixin(LitElement) implements ModalProps {
                 </div>
             </article>
             <footer class="c-modal-footer">
-                <pie-button
-                    variant="primary"
-                    type="submit"
-                    @click="${() => this._dialog?.close('leading')}"
-                    data-test-id="modal-leading-action">
-                    Confirm
-                </pie-button>
+                ${leadingAction ? this.renderLeadingAction() : nothing}
```

Comment:
`renderLeadingAction` can already return nothing so can we just have the function call here, or simplify the function instead?

Reply:
It would only return nothing if `text` isn't supplied otherwise it would return a button. The `nothing` on this particular line covers  the `leadingAction` prop. I could move it all into a method and just have a function here?

Reply:
Actually I think you're right about us needing `nothing` in both places, all good.